src/utils/sharedPlots/FrequencyPlot.jsx

```diff
@@ -0,0 +1,267 @@
+import React from 'react';
+import { Vega } from 'react-vega';
+import PropTypes from 'prop-types';
+
+const FrequencyPlot = (props) => {
+  const {
+    hierarchy, properties, config, actions,
+  } = props;
+
+  const generateSpec = () => {
+    let legend = [];
+
+    if (config.legend.enabled) {
+      legend = [
+        {
+          fill: 'color',
+          title: 'Cell Set',
+          titleColor: config.colour.masterColour,
+          type: 'symbol',
+          orient: config.legend.position,
+          offset: config.legend.offset,
+          symbolType: 'square',
+          symbolSize: { value: 200 },
+          encode: {
+            labels: {
+              update: {
+                text: {
+                  scale: 'c', field: 'label',
+                },
+                fill: { value: config.colour.masterColour },
+              },
+            },
+          },
+          direction: 'horizontal',
+          labelFont: { value: config.fontStyle.font },
+          titleFont: { value: config.fontStyle.font },
+        },
+      ];
+    }
+    return {
+      $schema: 'https://vega.github.io/schema/vega/v5.json',
+      width: config.dimensions.width,
+      height: config.dimensions.height,
+      autosize: { type: 'fit', resize: false },
+      background: config.colour.toggleInvert,
+      padding: 5,
+
+      data: [
+        {
+          name: 'data',
+          transform: [
+            {
+              type: 'stack',
+              groupby: ['x'],
+              sort: { field: 'c' },
+              field: 'y',
+            },
+          ],
+        },
+      ],
+
+      scales: [
```

Comment:
why did you move the plot Vega spec in here?

Reply:
To follow the logic used in ElbowPlot and the plots in configure embedding, where plots are in separate components.

If you are asking about why moving them to utils, then because since this plot is being used in different places then I felt it made more sense to have them in a shared space.

Reply:
Moved them to sharedPlots in /components.